pravega/pravega
# Issue 3113: Kubernetes based system test framework to deploy and test Pravega.

(Note: this PR should be merged only after PR: #3114  has been merged to master).
**Change log description**  
Ensure the Pravega tests can run on a K8 Cluster.  

**Purpose of the change**  
Fixes the following aspects of #3113 
- enable Pravega deployment using pravega operator and ensure related system tests are also run on the K8 cluster.
- enable `Service#scaleService( newinstanceCount)` api for bookkeeper, pravega-controller and pravega segment store.

**What the code does**  
The code ensures pravega related system tests can be run on a K8 cluster. 

**How to verify it**  
Pravega tests should run in a K8 Environment.

```
gradle --info startK8SystemTests -PpravegaVersion=<pravegaVersion> 
-PrepoUrl=<repo where the build artifacts are present> --tests io.pravega.test.system.ReaderCheckpointTest
```
Note: Validation is in progress.
